src/Corax/Indexing/IndexWriter.cs

```diff
@@ -629,7 +629,25 @@ private bool TryDeleteEntry(Slice termSlice, out long entryId)
             FlushBatch();
             return TryDeleteEntry(termSlice, out entryId);
         }
-        
+
+        public void DeleteByPrefix(string term)
+        {
+            using var __ = Slice.From(_transaction.Allocator, term, ByteStringType.Immutable, out var termSlice);
+
+            var fieldsTree = _fieldsTree.CompactTreeFor(_fieldsMapping.GetByFieldId(Constants.IndexWriter.PrimaryKeyFieldId).FieldName);
+
+            var iterator = fieldsTree.Iterate();
+            iterator.Seek(termSlice);
+
+            while (iterator.MoveNext(out var key, out var id, out _))
```

Comment:
We call `iterator.Seek()` and then `iterator.MoveNext()` immediately? Won't we skip the first item (the one which we seek to in the first place)? CC @maciejaszyk

Reply:
No, I've looked into `TermProvider.StartsWith` and for `forward` iterator is used in the same manner.